test/sort.jl

```diff
@@ -142,6 +143,78 @@ using DataFrames, Random, Test, CategoricalArrays
     end
 end
 
+@testset "correctness of checkunique keyword" begin
+    dv1 = [9, 1, 8, missing, 3, 3, 7, missing]
+    dv2 = [9, 1, 8, missing, 3, 3, 7, missing]
+    dv3 = Vector{Union{Int, Missing}}(1:8)
+    dv4 = 8:-1:1
+    cv1 = CategoricalArray(dv1, ordered=true)
+
+    d = DataFrame(dv1=dv1, dv2=dv2, dv3=dv3, dv4=dv4, cv1=cv1)
+
+    ## logic:
+    ### Test each every selector in the following order:
+    ### Symbol, String, Vect{ColumnIndex}, Order, Vect{ColIndex, Order}
+
+    # issorted
+    @test_throws ArgumentError issorted(d, :dv1, checkunique=true)
+    @test issorted(d, :dv3, checkunique=true)
+    @test issorted(d, "dv3", checkunique=true)
+    @test issorted(d, ["dv3", "dv4"], checkunique=true)
+    @test issorted(d, :dv4, rev = true, checkunique=true)
+    @test issorted(d, order(:dv4, rev=true), checkunique=true)
+    @test_throws ArgumentError issorted(d, order(:dv4, by=x -> -x), checkunique=true)
+    @test_throws ArgumentError issorted(d, order(:dv4, lt= >), checkunique=true)
+    @test issorted(d, [:dv3, order(:dv4, rev=true)], checkunique=true)
+    @test issorted(d, [:dv3, :dv4], rev = [false, true], checkunique=true)
+    @test_throws ArgumentError issorted(d, :dv3, by=x-> -x, checkunique=true)
+    @test_throws ArgumentError issorted(d, :dv3, lt = >, checkunique=true)
+    @test issorted(d, [order(:dv3, rev=false), order(:dv4, rev=true)], checkunique=true)
+    @test issorted(d, [order(:dv3, by=identity), order(:dv4, rev=true)], checkunique=true)
+
+    # sort
```

Comment:
Also test cases that do not throw errors? Overall it would make sense to use exactly the same tests for `issorted`, `sort`, `sort!` and `sortperm`, except for additional tests that apply only for some function(s).

Reply:
I agree. I added those to make sure every line of `issorted` is accounted as being tested so as not to decrease the coverage figure

Reply:
Coverage is the minimal requirement for developing tests. We need to make sure that we cover with the test every case. This is especially relevant as sorting functionality was developed a long time ago and might not have super precise test coverage.

Reply:
Initally I relied on the previous tests and just added the ones pertinent to `_perform_uniqueness_checks`. In other words, the tests to catch the issues with the possible cases of `cols` I didn't catch earlier. So that's why those tests don't error. If the `cols` parsing were broken, they would fail

Reply:
Do I understand it correctly that you say that you believe that your tests are providing a comprevensive coverage?

Reply:
Regarding the possible options for `cols`, I believed so. I might have missed something